test/index.test.js

```diff
@@ -408,8 +412,58 @@ test('Enable response status code validation for every route', async t => {
 
   t.assert.strictEqual(response.statusCode, 500)
   t.assert.deepStrictEqual(JSON.parse(response.payload), {
+    code: 'FST_NO_SCHEMA_DEFINED_ERROR',
     statusCode: 500,
     error: 'Internal Server Error',
     message: 'No schema defined for status code 200'
   })
 })
+
+describe('isResponseValidationError', () => {
+  test('Should identify that the error is emitted by this plugin', async t => {
+    const fastify = Fastify()
+    await fastify.register(plugin)
+
+    fastify.route({
+      method: 'GET',
+      url: '/',
+      schema: {
+        response: {
+          '2xx': {
+            type: 'object',
+            properties: {
+              answer: { type: 'number' }
+            }
+          }
+        }
+      },
+      handler: async (req, reply) => {
+        return { answer: '42' }
+      }
+    })
+
+    const errorHandler = mock.fn((error, request, reply) => {
+      t.assert.ok(plugin.isResponseValidationError(error))
+      return reply.status(200).send()
+    })
+
+    fastify.setErrorHandler(errorHandler)
+
+    await fastify.inject({
+      method: 'GET',
+      url: '/'
+    })
+
+    t.assert.equal(errorHandler.mock.callCount(), 1)
+    t.assert.strictEqual(errorHandler.mock.calls[0].error, undefined, 'errorHandler function has emitted an unexpected error')
```

Comment:
I want it to be as close as possible to how users are supposed to use this utility function, but I’m not sure this is the best way to test this case.